demos/coreMQTT/mqtt_demo_mutual_auth.c

```diff
@@ -447,7 +502,11 @@ int RunCoreMqttMutualAuthDemo( bool awsIotMqttMode,
      */
     ulGlobalEntryTimeMs = prvGetTimeMs();
 
-    for( ; ulDemoRunCount < democonfigMQTT_MAX_DEMO_COUNT; ulDemoRunCount++ )
+    /* Seed randon number generator for use with retry utils library for
+     * backoff calculation during failed connection retry attempts. */
+    xDemoStatus = prvSeedRandomNumberGenerator();
+
+    for( ; ( ( xDemoStatus == pdPASS ) && ( ulDemoRunCount < democonfigMQTT_MAX_DEMO_COUNT ) ); ulDemoRunCount++ )
```

Comment:
( xDemoStatus == pdPASS ) in this condition will fail the retries we are doing. Please move it out of this for loop condition.

Reply:
Have introduced a new boolean variable that represents only whether the random number generation was successful.